server/serverimpl_test.go

```diff
@@ -72,6 +73,15 @@ func TestServerStartStopWithCancel(t *testing.T) {
 	assert.ErrorIs(t, err, context.Canceled)
 }
 
+func TestServerStartStopCloseNoLeakGoroutine(t *testing.T) {
+	goleak.VerifyNone(t)
```

Comment:
```suggestion
	defer goleak.VerifyNone(t)
```
https://pkg.go.dev/go.uber.org/goleak#VerifyNone